examples/demo-nft-module/README.md

```diff
@@ -0,0 +1,374 @@
+# How to create new module for Sovereign Rollup
+
+
+# Getting Started
+
+Module is a separate crate that needs to implement `sov_modules_api::Module` trait and it's own change logic based on input messages.
+
+Let's start with defining basic dependencies in `Cargo.toml`:
+
+```toml
+[dependencies]
+anyhow = { workspace = true }
+sov-modules-api = { path = "../../module-system/sov-modules-api", default-features = false }
+sov-modules-macros = { path = "../../module-system/sov-modules-macros" }
+```
+
+## Module definition
+
+NFT module is defined as following:
+
+```rust
+use sov_modules_api::Context;
+use sov_modules_macros::ModuleInfo;
+
+#[derive(ModuleInfo, Clone)]
+pub struct NonFungibleToken<C: Context> {
+    #[address]
+    pub address: C::Address,
+
+    #[state]
+    pub(crate) admin: sov_state::StateValue<C::Address>,
+
+    #[state]
+    pub(crate) owners: sov_state::StateMap<u64, C::Address>,
+}
+```
+
+Where it has:
+
+1. Address. Each module has to have address. TBD: Why, and how it is defined and used.
+2. Several state attributes. In this case it is address of admin, and map of token id to owner address. Token id is u64 for simplicity.
+
+TBD: What is context, what does it hold?
```

Comment:
This question

Reply:
Context is a trait which allows the runtime to pass verified data to modules during execution. Currently the only required method in Context `sender()` which returns the address of the person who initiated the transaction (the signer).

Context also inherits the `Spec` trait, which defines the concrete types used by the rollup for `Hash`ing, persistant data `Storage`, digital `Signature`s, and `Addresses`. The `Spec` trait allows rollups to easily tailor themselves to different Zkvms. By being generic over a `Spec`, a rollup can ensure that any potentially SNARK-unfriendly cryptography can be easily swapped out.